content/posts/jenkins-snyk.md

```diff
@@ -0,0 +1,357 @@
+
+
+```groovy
+pipeline {
+  agent {
+    kubernetes {
+      label 'python-dev'
+      yaml """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: snyk-python
+    image: snyk/snyk-cli:python-3
+    command:
+    - /bin/cat
+    tty: true
+  - name: snyk-docker
+    image: snyk/snyk-cli:docker
+    command:
+    - /bin/cat
+    tty: tru
+    env:
+    - name: DOCKER_HOST
+      value: tcp://localhost:2375
+  - name: dind
+    image: docker:18.09-dind
+    securityContext:
+      privileged: true
+    volumeMounts:
+      - name: dind-storage
+        mountPath: /var/lib/docker
+  volumes:
+  - name: dind-storage
+    emptyDir: {}
+"""
+    }
+  }
+  stages {
+    stage('Snyk Scan') {
+      failFast true
+      environment {
+        SNYK_TOKEN = credentials('snyk-token')
+      }	
+      parallel {
+        stage('dependency scan') {
+          steps {
+            container('snyk-python') {
+              sh """
+                pip install -r requirements.txt
+                snyk auth ${SNYK_TOKEN}
+                snyk test --json \
+                  --file=requirements.txt \
+                  --severity-threshold=high \
+                  --org=cvega \
+                  --project-name=project-python
+              """
+            }
+          }
+        }
+        stage('docker scan') {
+          steps {
+            container('snyk-docker') {
+              sh """
+                docker build -t python-project .
+                snyk auth ${SNYK_TOKEN}
+                snyk test --json \
+                  --docker python-project:latest \
+                  --file=Dockerfile \
+                  --severity-threshold=high \
+                  --org=cvega \
+                  --project-name=project-python
+              """
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+#### 5.  **Snyk Reports and Wizard**
+
+Like Jenkins and CloudBees, Snyk has quite a few valuable features. The Snyk portal provides data about  vulnerabilities in your project.
+
+![snyk portal](/img/jenkins-snyk/snyk-web.png)
+
+The CLI wizard feature of Snyk enables developers to take action and update dependencies. Using multi-branch jobs this makes testing updates to dependencies manageable.
+
+![snyk wizard](/img/jenkins-snyk/snyk-wizard.png)
+
+#### 6. **Summary**
+
+We've presented several different ways to integrate Snyk and Jenkins using different types of technology and all of it fits together well. This is exactly the type of testing that could have prevented recent data breaches. This is also the type of development self-service that actually empowers developers and security teams to take the proper mitigation steps.
+
+**Last but not Least**
+
+Join CloudBees and Snyk at DevOps World.
+
+![DevOps World - Jenkins World](/img/jenkins-snyk/dwjw.png)
```

Comment:
One more thing - would be nice to link this image to https://www.cloudbees.com/devops-world/